src/dataviews/dataview-model-base.js

```diff
@@ -336,6 +339,24 @@ module.exports = Model.extend({
     if (!(source instanceof AnalysisModel)) {
       throw new Error('Source must be an instance of AnalysisModel');
     }
+  },
+
+  _checkBBoxFilter: function () {
+    if (this.syncsOnBoundingBoxChanges() && !this._bboxFilter) {
+      throw new Error('Cannot sync on bounding box changes. There is no bounding box filter.');
```

Comment:
This will always be present in the context of BUILDER right? I guess that's not the case when using this through the API.

Reply:
Yes. `dataviews-factory` is the responsible for creating dataviews and it's providing the bboxFilter from the very beginning.

I opted to throw an Error in case the users want the dataview to be synced but there are no bboxFilter, since they won't get what they expect.